controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -173,20 +175,25 @@ protected void startUp() {
                 monitor.startAsync();
             }
 
-            ClientConfig clientConfig = ClientConfig.builder()
-                                                    .controllerURI(URI.create((serviceConfig.getGRPCServerConfig().get().isTlsEnabled() ?
-                                                                          "tls://" : "tcp://") + "localhost"))
-                                                    .trustStore(serviceConfig.getGRPCServerConfig().get().getTlsTrustStore())
-                                                    .validateHostName(false)
-                                                    .build();
+            // This client config is used by the segment store helper (SegmentHelper) to connect to the segment store.
+            ClientConfig.ClientConfigBuilder clientConfigBuilder = ClientConfig.builder()
+                    .controllerURI(URI.create((serviceConfig.getGRPCServerConfig().get().isTlsEnabled() ?
+                            "tls://" : "tcp://") + "localhost"))
+                    .trustStore(serviceConfig.getGRPCServerConfig().get().getTlsTrustStore())
+                    .validateHostName(false);
 
-            connectionFactory = new ConnectionFactoryImpl(clientConfig);
+            Optional<Boolean> tlsEnabledForSegmentStore = BooleanUtils.extract(serviceConfig.getTlsEnabledForSegmentStore());
+            if (tlsEnabledForSegmentStore.isPresent()) {
+                clientConfigBuilder.enableTlsToSegmentStore(tlsEnabledForSegmentStore.get());
+            }
+
+            connectionFactory = new ConnectionFactoryImpl(clientConfigBuilder.build());
             segmentHelperRef.compareAndSet(null, new SegmentHelper(connectionFactory, hostStore));
 
             GrpcAuthHelper authHelper = new GrpcAuthHelper(serviceConfig.getGRPCServerConfig().get().isAuthorizationEnabled(),
                     serviceConfig.getGRPCServerConfig().get().getTokenSigningKey(),
                     serviceConfig.getGRPCServerConfig().get().getAccessTokenTTLInSeconds());
-            
+
```

Comment:
nit: revert

Reply:
The blank line is the same as before, but there are new lines added to the code above the `GrpcAuthHelper` instantiation, which might be causing this difference. The formatting looks OK in the fil when viewing it standalone:

https://github.com/pravega/pravega/blob/ea1aadd4920cc27aa4e59b0d318ffd90c117edf9/controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java#L196

Am I missing something?